kong/router/atc.lua

```diff
@@ -553,14 +634,29 @@ function _M:exec(ctx)
     headers_key = get_headers_key(headers)
   end
 
+  local queries, queries_key
+  if self.match_queries then
+    local err
+    queries, err = get_uri_args()
+    if err == "truncated" then
+      local lua_max_uri_args = kong and kong.configuration and kong.configuration.lua_max_uri_args or 100
+      ngx_log(ngx_ERR, "router: not all request queries were read in order to determine the route as ",
+                       "the request contains more than ", lua_max_uri_args, " queries, route selection ",
+                       "may be inaccurate, consider increasing the 'lua_max_uri_args' configuration value ",
+                       "(currently at ", lua_max_uri_args, ")")
+    end
+
+    queries_key = get_queries_key(queries)
+  end
+
   req_uri = strip_uri_args(req_uri)
 
   -- cache lookup
 
   local cache_key = (req_method or "") .. "|" ..
                     (req_uri    or "") .. "|" ..
                     (req_host   or "") .. "|" ..
-                    (sni        or "") .. (headers_key or "")
+                    (sni        or "") .. (headers_key or "") .. (queries_key or "")
```

Comment:
I think we should have a seprator `|` between these components.

```suggestion
                    (sni        or "") .. "|" ..  (headers_key or "") .. "|" .. (queries_key or "")
```

Otherwise we might consider:

`/?a=b&c=d` the same as

```
A: b
C: d
```

which should definitely not be considered as the same request.

Reply:
Change to `(headers_key or "|") .. (queries_key or "|")`, do think it is OK?

Reply:
No, we need separator between the cache key components just as all other part does it.

I think your proposed change could work, but it is inconsistent with the above code.

Reply:
`get_headers_key()` and `get_queries_key()` have added the seperator for `headers_key ` and `queries_key `.
The code is:
```
str_buf:putf("|%s=%s", name, value)
```
So if `headers_key` is nil, the cache key will be `...||a=b,c=d`.

Reply:
done